survey_posix_test.go

```diff
@@ -28,7 +28,6 @@ func RunTest(t *testing.T, procedure func(*expect.Console), test func(terminal.S
 	}()
 
 	err = test(Stdio(c))
-	require.Nil(t, err)
```

Comment:
Yes, because, L#30 would return an error when SIGINT was encountered, and the tests would fail (due to the assert). So I shifted the error assertion to the helper functions (`RunPromptTest` and the other one).

I could revert this, but that would mean rewriting the `RunTest` function exclusively for the OnInterrupt test, which isn't a very good idea imo.